packages/targets/dydx-stark/src/transport/send.ts

```diff
@@ -0,0 +1,96 @@
+import { Transport, TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { ResponseCache } from '@chainlink/external-adapter-framework/cache/response'
+import { Requester } from '@chainlink/external-adapter-framework/util/requester'
+import {
+  AdapterRequest,
+  AdapterResponse,
+  makeLogger,
+} from '@chainlink/external-adapter-framework/util'
+import { BaseEndpointTypes, DyDxResponse, inputParameters } from '../endpoint/send'
+import { getPricePayload, PriceDataPoint, PriceStarkPayload, requireNormalizedPrice } from './utils'
+import { calculateHttpRequestKey } from '@chainlink/external-adapter-framework/cache'
+
+const logger = makeLogger('DyDxTransport')
+
+export type DyDxTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: PriceStarkPayload
+    ResponseBody: DyDxResponse
+  }
+}
+
+export class DyDxTransport implements Transport<DyDxTransportTypes> {
+  name!: string
+  responseCache!: ResponseCache<DyDxTransportTypes>
+  requester!: Requester
+
+  async initialize(
+    dependencies: TransportDependencies<DyDxTransportTypes>,
+    _adapterSettings: DyDxTransportTypes['Settings'],
+    _endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    this.responseCache = dependencies.responseCache
+    this.requester = dependencies.requester
+    this.name = transportName
+  }
+
+  async foregroundExecute(
+    req: AdapterRequest<typeof inputParameters.validated>,
+    config: DyDxTransportTypes['Settings'],
+  ): Promise<AdapterResponse<DyDxTransportTypes['Response']>> {
+    const { asset, result } = req.requestContext.data
+
+    const priceData: PriceDataPoint = {
+      oracleName: config.ORACLE_NAME,
+      assetName: asset,
+      // Current timestamp in seconds
+      timestamp: Math.floor(Date.now() / 1000),
+      price: requireNormalizedPrice(result),
+    }
+
+    const payload = await getPricePayload(config.PRIVATE_KEY, config.STARK_MESSAGE, priceData)
+
+    const requestConfig = {
+      baseURL: config.API_ENDPOINT,
+      url: '',
+      method: 'POST',
+      data: payload,
+    }
+
+    logger.debug('Sending payload: ', { payload })
+
+    const providerDataRequestedUnixMs = Date.now()
+
+    const providerResponse = await this.requester.request<DyDxResponse>(
+      calculateHttpRequestKey<DyDxTransportTypes>({
+        context: {
+          adapterSettings: config,
+          inputParameters,
+          endpointName: req.requestContext.endpointName,
+        },
+        data: payload,
+        transportName: this.name,
+      }),
+      requestConfig,
+    )
+
+    const response = {
+      data: {
+        result: providerResponse.response.data,
+      },
+      statusCode: 200,
+      // see the endpoint file for explanation why the type cast is needed
+      result: providerResponse.response.data as unknown as string,
```

Comment:
Since the expected response is `{ market: string; price: string }`. Should we return the price here?
```suggestion
      result: providerResponse.response.data.price,
```

Reply:
I wanted to keep the same logic here. The current EA also returns the whole response. Furthermore i'm not sure if `{ market: string; price: string }` is the right or complete response. I took it from current readme's output example (i wasn't able to actually get successful response).

Reply:
Normally we just return null for complex results like in PoR but not sure if this field is used. I think it'd conflict with v3 expectations though so we should keep the full response in the `data` field but just the price in the external `result`. If the purpose of the adapter is to post the answer, the response from the adapter may not be too important other than the status code. Also I think we can trust the README for now since we're sort of going into this migration blind